python/common/org/python/types/Str.java

```diff
@@ -309,11 +320,21 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
                     }
 
                     if (step == 1) {
-                        sliced = this.value.substring((int) start, (int) stop);
+                        if (start < stop) {
+                            sliced = this.value.substring((int) start, (int) stop);
+                        } else {
+                            sliced = "";
+                        }
                     } else {
                         java.lang.StringBuffer buffer = new java.lang.StringBuffer();
-                        for (long i = start; i < stop; i += step) {
+                        long i = start;
+                        while (true) {
```

Comment:
Would it be clearer to have a `while (condition)` instead of following `while (true)` with an if block to break?

Also, I'm just curious, is there a specific reason to switch from a for loop to a while loop? We could essentially express the same logic as the condition in the for loop, and the logic is still counting steps away from start.